flow/flow.go

```diff
@@ -0,0 +1,303 @@
+package flow
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/micro/go-micro/v2/broker"
+	"github.com/micro/go-micro/v2/client"
+	"github.com/micro/go-micro/v2/logger"
+)
+
+type Flow interface {
+	// Init flow with options
+	Init(...Option) error
+	// Get flow options
+	Options() Options
+	// Create step in specific flow
+	CreateStep(flow string, step *Step) error
+	// Delete step from specific flow
+	DeleteStep(flow string, step *Step) error
+	// Replace step in specific flow
+	ReplaceStep(flow string, oldstep *Step, newstep *Step) error
+	// Lookup specific flow
+	Lookup(flow string) ([]*Step, error)
+	// Execute specific flow and returns request id and error, optionally fills rsp
+	Execute(flow string, req interface{}, rsp interface{}, opts ...ExecuteOption) (string, error)
+	// Resume specific paused flow execution by request id
+	Resume(flow string, reqID string) error
+	// Pause specific flow execution by request id
+	Pause(flow string, reqID string) error
+	// Abort specific flow execution by request id
+	Abort(flow string, reqID string) error
+	// State show state specific flow execution by request id
+	State(flow string, reqID string) (string, error)
+	// Result get result of the flow step
+	Result(flow string, reqID string, step *Step) ([]byte, error)
+	// Stop executor and drain active workers
+	Stop() error
+}
+
+type Step struct {
+	// name of step
+	ID string
+	// Retry count for step
+	Retry int
+	// Timeout for step
+	Timeout int
+	// Step operation to execute
+	Operation Operation
+	// Which step use as input
+	Input string
```

Comment:
What is an input and output?

Reply:
Its step names where put/get data in data store.
So if steps not specify input, we thinks that all steps get the same input that passed via execute to flow. If steps have different input - it can get other steps output as input.

Reply:
Its unclear even still. I have zero idea what to put in the Input/Output. The string values are not useful unless I know how to use them, especially for something so opaque.

Reply:
So i can change it to *Step , but from it flow uses only id to determine store key to retrieve data